### PR TITLE
Slider

### PR DESCRIPTION
**Pull request by qroll**
**Changes**
New slider input to select from a range of numeric values. The ranged variant is intended to be used in the histogram

- [delete] branch

**Changelog entry**

- New `InputSlider` and `InputRangeSlider`

**Additional information**

- [Figma](https://www.figma.com/file/us90jOpWBahsqK2VAluPvD/Flagship-Design-System?type=design&node-id=21395-21195&mode=design&t=5drpfCc3SvHCpodI-0)

### CODE REVIEW

File: src/input-slider/input-slider.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { InputRangeSlider } from "../input-range-slider";
+import { InputSliderProps } from "./types";
+
+export const InputSlider = ({
+    value,
+    ariaLabel,
+    onChange,
+    ...otherProps
+}: InputSliderProps) => {
+    const [selection, setSelection] = useState<number[]>(initialiseSelection());
```
Comment (keithtxw) [inline src/input-slider/input-slider.tsx:11]: We're missing section headers

File: stories/form/form-range-slider/props-table.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                defaultValue: "2",
+            },
+            {
+                name: "minRange",
+                description: "The minimum difference between values",
+                propTypes: ["number"],
+            },
+            {
+                name: "disabled",
+                description: "the number of controls",
```
Comment (keithtxw) [inline stories/form/form-range-slider/props-table.tsx:47]: This looks incorrect

File: stories/form/form-range-slider/props-table.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                propTypes: ["boolean"],
+            },
+            {
+                name: "readOnly",
+                description: "the number of controls",
```
Comment (keithtxw) [inline stories/form/form-range-slider/props-table.tsx:52]: Incorrect description

File: stories/form/form-range-slider/props-table.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                propTypes: ["boolean"],
+            },
+            {
+                name: "colors",
+                description: "List of custom colors for each track segment",
```
Comment (keithtxw) [inline stories/form/form-range-slider/props-table.tsx:57]: Perhaps want to add the requirement that the expected length is (number of values + 1)?

File: stories/form/form-slider/props-table.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            {
+                name: "step",
+                description:
+                    "The interval between the previous and next values",
+                propTypes: ["number"],
+                defaultValue: "1",
+            },
+            {
+                name: "disabled",
+                description: "the number of controls",
```
Comment (keithtxw) [inline stories/form/form-slider/props-table.tsx:36]: `disabled` and `readOnly` have incorrect descriptions

File: stories/form/form-slider/props-table.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import { ApiTableSectionProps } from "../../storybook-common/api-table/types";
+import { SHARED_FORM_PROPS_DATA } from "../shared-props-data";
+
+const DATA: ApiTableSectionProps[] = [
+    {
+        name: "InputSlider specific props",
+        attributes: [
+            {
+                name: "value",
+                description: "The input values",
```
Comment (keithtxw) [inline stories/form/form-slider/props-table.tsx:12]:
```suggestion
                description: "The input value",
```

**Review by keithtxw — CHANGES_REQUESTED** [review]

**Comment by qroll** [comment]
Updated incorrect prop descriptions and also added documentation for `className`/`data-testid`/`id` and `ariaLabels`

**Comment by qroll** [comment]
force pushed to rebase changes on top of master

File: stories/form/form-slider/props-table.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            },
+            {
+                name: "readOnly",
+                description:
+                    "Indicates if the component is in the readonly state",
+                propTypes: ["boolean"],
+            },
+            {
+                name: "colors",
+                description: "List of custom colors for each track segment",
```
Comment (keithtxw) [inline stories/form/form-slider/props-table.tsx:63]: To add the same description as the one in `form-range-slider` props table

**Review by keithtxw — CHANGES_REQUESTED** [review]

**Review by keithtxw — APPROVED** [review]

**Pull request merged**